test/e2e/kubernetes/kubernetes_test.go

```diff
@@ -120,6 +124,17 @@ var _ = BeforeSuite(func() {
 		Expect(success).To(BeTrue())
 		firstMasterRegexp, err = regexp.Compile(firstMasterRegexStr)
 		Expect(err).NotTo(HaveOccurred())
+		if hasAddon, addon := eng.HasAddon("cluster-autoscaler"); hasAddon {
+			if len(addon.Pools) > 0 {
+				for _, pool := range addon.Pools {
+					p := eng.ExpandedDefinition.Properties.GetAgentPoolIndexByName(pool.Name)
+					maxNodes, _ := strconv.Atoi(pool.Config["max-nodes"])
+					if maxNodes > eng.ExpandedDefinition.Properties.AgentPoolProfiles[p].Count {
+						clusterAutoscalerEngaged = true
```

Comment:
nit: we can break here, no need to check for the rest of the pools once we've found one

Reply:
thx!